### PR TITLE
feat: allow default cookie options to be overridden

### PR DESCRIPTION
**Pull request by stevesweets**
In my use-case, i'm in an iframe with a strict security policy i cannot change.

This results in `lax` cookies being blocked.

However, `secure`, `sameSite: none` cookies are not, which resolves this problem.

Seemed a simple enough change for me to just do a pr for.

Thanks for the great work on the package.

#### Checklist

- [x] run `npm run test` and ~~`npm run benchmark`~~
- [x] tests and/or ~~benchmarks are included~~
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

#### Test results

```
➜  fastify-oauth2 git:(master) npm run test

> @fastify/oauth2@7.2.2 test
> npm run test:unit && npm run test:typescript


> @fastify/oauth2@7.2.2 test:unit
> tap

​ PASS ​ test/index.test.js 112 OK 138.59ms
​

                         
  🌈 SUMMARY RESULTS 🌈  
                         
​
Suites:   ​1 passed​, ​1 of 1 completed​
Asserts:  ​​​112 passed​, ​of 112​
​Time:​   ​543.749ms​
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |     100 |      100 |     100 |     100 |                   
 index.js |     100 |      100 |     100 |     100 |                   
----------|---------|----------|---------|---------|-------------------
```

### CODE REVIEW

**Comment by stevesweets** [comment]
I struck out `npm run benchmark` since this isn't a provided script in package.json.

File: index.js [hunk 1]
```diff
@@ -78,14 +78,15 @@ function fastifyOauth2 (fastify, options, next) {
   const startRedirectPath = options.startRedirectPath
   const tags = options.tags || []
   const schema = options.schema || { tags }
+  const cookieOpts = options.cookie || {
```
Comment (Uzlopak) [inline index.js:81]: Can we have something like
```suggestion
  const cookieOpts = Object.assign({}, options.cookie, {
```

Comment (stevesweets) [reply index.js:81]: Sure|

**Comment by Uzlopak** [comment]
Missing typings
Missing unit tests
Missing typings tests

File: index.js [hunk 2]
```diff
@@ -78,14 +78,15 @@ function fastifyOauth2 (fastify, options, next) {
   const startRedirectPath = options.startRedirectPath
   const tags = options.tags || []
   const schema = options.schema || { tags }
+  const cookieOpts = Object.assign({}, options.cookie, {
+    httpOnly: true,
+    sameSite: 'lax'
+  }
```
Comment (Uzlopak) [inline index.js:84]:
```suggestion
  })
```

**Comment by stevesweets** [comment]
Updated. Hopefully what you're looking for  👍

**Comment by jorisdugue** [comment]
Hey, maybe we should add a default **priority** on cookie to medium?

For more information 😉 :
https://bugs.chromium.org/p/chromium/issues/detail?id=232693

**Review by Uzlopak — APPROVED** [review]

**Pull request merged**